### PR TITLE
Add ZNCZ11LM

### PR DESCRIPTION
**Pull request by xfzha0**
ZNCZ11LM is very few on the market, I bought two.
Now its functions can work, but the led_disable_night time is not normal. I am here in China. Its effective time is not 9:00 PM - 9:00 AM . It is delayed for 8 hours, from 5:00AM to 5:00 PM.
Do you have a way to fix this problem?

### CODE REVIEW

File: converters/toZigbee.js [hunk 1]
```diff
@@ -1717,6 +1717,36 @@ const converters = {
             }
         },
     },
+    xiaomi_ZNCZ11LM_power_outage_memory: {
```
Comment (Koenkk) [inline converters/toZigbee.js:1720]: Please merge this converter with https://github.com/Koenkk/zigbee-herdsman-converters/blob/0a757bf9787a2cbcaccef27a178beb260966b6a9/converters/toZigbee.js#L1723

File: converters/toZigbee.js [hunk 2]
```diff
@@ -1717,6 +1717,36 @@ const converters = {
...
+        key: ['power_outage_memory'],
+        convertSet: async (entity, key, value, meta) => {
+            const payload = value ?
+                [0xaa, 0x80, 0x05, 0xd1, 0x47, 0x00, 0x01, 0x10, 0x01] :
+                [0xaa, 0x80, 0x05, 0xd1, 0x47, 0x01, 0x01, 0x10, 0x00];
+            await entity.write('genBasic', {0xFFF0: {value: payload, type: 0x41}}, manufacturerOptions.xiaomi);
+            return {state: {power_outage_memory: value}};
+        },
+    },
+    xiaomi_ZNCZ11LM_led_disabled_night: {
```
Comment (Koenkk) [inline converters/toZigbee.js:1730]: Please merge this converter with https://github.com/Koenkk/zigbee-herdsman-converters/blob/0a757bf9787a2cbcaccef27a178beb260966b6a9/converters/toZigbee.js#L1771

File: converters/toZigbee.js [hunk 3]
```diff
@@ -1717,6 +1717,36 @@ const converters = {
...
+        key: ['led_disabled_night'],
+        convertSet: async (entity, key, value, meta) => {
+            const payload = value ?
+                [0xaa, 0x80, 0x05, 0xd1, 0x47, 0x00, 0x03, 0x10, 0x00] :
+                [0xaa, 0x80, 0x05, 0xd1, 0x47, 0x01, 0x03, 0x10, 0x01];
+            await entity.write('genBasic', {0xFFF0: {value: payload, type: 0x41}}, manufacturerOptions.xiaomi);
+            return {state: {led_disabled_night: value}};
+        },
+    },
+    xiaomi_ZNCZ11LM_auto_off: {
```
Comment (Koenkk) [inline converters/toZigbee.js:1740]: Please merge with https://github.com/Koenkk/zigbee-herdsman-converters/blob/0a757bf9787a2cbcaccef27a178beb260966b6a9/converters/toZigbee.js#L1760

**Comment by xfzha0** [comment]
Okay, I will try to modify it.  
Do you have any ideas about led_disabled_night?

**Comment by Koenkk** [comment]
If you start zigbee2mqtt with `DEBUG=zigbee-herdsman:controller*`, do you occasionally (maybe try to force it by removing it from power for a few seconds) see a read on the `genTime` cluster?

**Comment by xfzha0** [comment]
<code> zigbee-herdsman:controller:endpoint ReadResponse 0x00158d00015622fa/1 genTime({"time":680781430}, {"sendWhenActive":false,"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":1,"srcEndpoint":1,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) +8s</code>

**Comment by Koenkk** [comment]
@xfzha0 how are you running z2m? What is the output of the `date` command, is it correct? (if you are running in docker execute the command in the docker container).

**Comment by xfzha0** [comment]
Running Zigbee2MQTT in Virtual Environment.
Raspiberry 4b .
<code>
pi@raspberrypi:/opt/zigbee2mqtt/data $ date
Wed 28 Jul 23:33:54 CST 2021
</code>

**Comment by Koenkk** [comment]
@xfzha0 in your OP:

>  I am here in China.

But CST is not a Chinese timezone. Are you sure the time is correct?

**Comment by xfzha0** [comment]
@Koenkk 
Hmmmm..CST include China, Is it right? (China Standard Time).

**Comment by Koenkk** [comment]
Ah that makes sense, that timezone is exactly +8 hours ahead of UTC.

`genTime({"time":680781430},` -> the time here is the seconds since 1 January 2000 UTC. I think this device incorrectly uses the Zigbee spec and expects this to be the seconds since 1 January 2000 in the local time zone.

I've added an option to disable the time response in herdsman, then we can handle it ourselves in the `onEvent`

- First update to the latest z2m dev
- Then try with the following converter:
```js
    {
        zigbeeModel: ['lumi.plug.aq1'],
        model: 'ZNCZ11LM',
        vendor: 'Xiaomi',
        description: 'Aqara power plug ZigBee',
        fromZigbee: [fz.on_off, fz.xiaomi_power, fz.xiaomi_switch_basic],
        toZigbee: [tz.on_off, tz.xiaomi_power, tz.xiaomi_ZNCZ11LM_led_disabled_night,
            tz.xiaomi_ZNCZ11LM_power_outage_memory, tz.xiaomi_ZNCZ11LM_auto_off],
        exposes: [e.switch(), e.power().withAccess(ea.STATE_GET), e.energy(), e.temperature(),
            e.voltage().withAccess(ea.STATE), e.power_outage_memory().withAccess(ea.STATE_SET),
            e.led_disabled_night().withAccess(ea.STATE_SET),
            exposes.binary('auto_off', ea.STATE_SET, true, false)],
        onEvent: async (type, data, device) => {
            device.skipTimeResponse = true;
            if (type === 'message' && data.type === 'read' && data.cluster === 'genTime') {
                const OneJanuary2000 = new Date('January 01, 2000 00:00:00 UTC+00:00').getTime();
                const timeUTC = Math.round(((new Date()).getTime() - OneJanuary2000) / 1000);
                const localTime = timeUTC - (new Date()).getTimezoneOffset() * 60;
                device.getEndpoint(1).readResponse('genTime', data.meta.zclTransactionSequenceNumber, {time: localTime});
                console.log("\n\nCODE TRIGGERED\n\n");
            }
        },
    },
```

**Comment by xfzha0** [comment]
WOW, the <code>led_disabled_night</code>now it works normally.

**Comment by Koenkk** [comment]
@xfzha0 good, as final check, do you indeed see `console.log("\n\nCODE TRIGGERED\n\n");` logged now? (to confirm this code is actually triggered)

**Comment by xfzha0** [comment]
@Koenkk  Yes, I see in the console.log \n\nCODE TRIGGERED\n\n
CODE TRIGGERED

**Pull request merged**